### PR TITLE
bli/ready-file

### PR DESCRIPTION
**Pull request by brianrli**
ready-file flag

### CODE REVIEW

**Comment by rudle** [comment]
I can't remember: did we decide it was OK to change the command line invocation of `p2-exec` in our runit files?

**Comment by mpuncel** [comment]
I do believe it's okay, and we've done it plenty of times in the past e.g. by adding --extra-env

File: bin/p2-exec/main.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ var (
 		"Specifies an extra environment KEY=VALUE pair to set for the process. May be used multiple times. Takes precedence over --env if there are conflicts",
 	).StringMap()
 	launchableName = kingpin.Flag("launchable", "The key in $PLATFORM_CONFIG_PATH containing the cgroup parameters.").Short('l').String()
+	readyFile      = kingpin.Flag("ready-file", "Checking for the presence of a ready-to-run file before execing its argument").String()
```
Comment (mpuncel) [inline bin/p2-exec/main.go]: s/Checking/Check

File: bin/p2-exec/main.go [hunk 2]
```diff
@@ -110,6 +111,12 @@ func main() {
 		}
 	}
 
+	if *readyFile != "" {
+		if _, err := os.Stat(*readyFile); os.IsNotExist(err) {
```
Comment (bwester) [inline bin/p2-exec/main.go]: Should the service start if `Stat()` returns, say, an I/O error?

File: bin/p2-exec/main.go [hunk 3]
```diff
@@ -32,6 +32,7 @@ var (
 		"Specifies an extra environment KEY=VALUE pair to set for the process. May be used multiple times. Takes precedence over --env if there are conflicts",
 	).StringMap()
 	launchableName = kingpin.Flag("launchable", "The key in $PLATFORM_CONFIG_PATH containing the cgroup parameters.").Short('l').String()
+	readyFile      = kingpin.Flag("ready-file", "Check for the presence of a ready-to-run file before execing its argument").String()
```
Comment (bwester) [inline bin/p2-exec/main.go]: nit: Calling this a "ready file" seems strange. To me, without the context you have right now, the name doesn't convey the meaning of what it does. I would suggest something more straightforward that doesn't introduce a new term:
```
  --require-file FILENAME
      If given, require the presence of the named file before running the command.
```

**Comment by bwester** [comment]
Oh, and (nit) please structure your commit message according to GitHub's and git's [recommended format](https://github.com/erlang/otp/wiki/Writing-good-commit-messages). Keep the title separate from the message.

**Pull request closed without merging**